README.md

```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
+            "10.1.1.21",
+            "10.1.1.22:5703"
+        ],
+        "smartRouting": true,
+        "redoOperation": true,
+        "connectionTimeout": 6000,
+        "connectionAttemptPeriod": 5000,
+        "connectionAttemptLimit": 5
+    }
+}
+```
+
+### Programmatic Client Network Configuration
+
+Here is an example of configuring network for Node.js Client programmatically.
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.addresses.push('10.1.1.21', '10.1.1.22:5703');
+clientConfig.networkConfig.smartRouting = true;
+clientConfig.networkConfig.redoOperation = true;
+clientConfig.networkConfig.connectionTimeout = 6000;
+clientConfig.networkConfig.connectionAttemptPeriod = 5000;
+clientConfig.networkConfig.connectionAttemptLimit = 5;
+```
+
+## 1. Providing the Member Addresses
+
+Address list is the initial list of cluster addresses to which the client will connect. The client uses this
+list to find an alive member. Although it may be enough to give only one address of a member in the cluster
+(since all members communicate with each other), it is recommended that you give the addresses for all the members.
+
+**Declarative:**
+
+```json
+{
+    "network": {
+        "clusterMembers": [
+            "10.1.1.21",
+            "10.1.1.22:5703"
+        ]
+    }
+}
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.addresses.push('10.1.1.21', '10.1.1.22:5703');
+```
+
+If the port part is omitted, then 5701, 5702 and 5703 will be tried in random order.
+
+You can provide multiple addresses with ports provided or not, as seen above. The provided list is shuffled and tried in random order. Its default value is `localhost`.
+
+## 2. Setting Smart Routing
+
+Smart routing defines whether the client mode is smart or unisocket. The following are example configurations.
+
+**Declarative:**
+
+```json
+{
+    "network": {
+        "smartRouting": true
+    }
+}
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.smartRouting = true;
+```
+
+Its default value is `true` (smart client mode).
+
+## 3. Enabling Redo Operation
+
+It enables/disables redo-able operations. While sending the requests to related members, operations can fail due to various reasons. Read-only operations are retried by default. If you want to enable retry for the other operations, you can set the `redoOperation` to `true`.
+
+**Declarative:**
+
+```json
+{
+    "network": {
+        "redoOperation": true
+    }
+}
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.redoOperation = true;
+```
+
+Its default value is `false` (disabled).
+
+## 4. Setting Connection Timeout
+
+Connection timeout is the timeout value in milliseconds for members to accept client connection requests. The following are the example configurations.
```

Comment:
What happens if the server does not respond within connection timeout? Does client quit trying altogether?

Reply:
I referenced Java documentation. See http://docs.hazelcast.org/docs/3.10.4/manual/html-single/index.html#setting-connection-timeout

Reply:
Furkan, if you know the answer to this, you should either include the link, or the actual content here which is specific to Node.js.

Reply:
I will add the answer.